src/lib/logs/dynatrace_client.py

```diff
@@ -54,18 +59,40 @@ def send_logs(context: LogsContext, logs: List[Dict]):
             )
             if status > 299:
                 context.error(f'Log ingest error: {status}, reason: {reason}, url: {log_ingest_url}, body: "{response}"')
-                number_of_http_errors += 1
+                if status == 400:
+                    self_monitoring.dynatrace_connectivity.append(DynatraceConnectivity.InvalidInput)
+                elif status == 401:
+                    self_monitoring.dynatrace_connectivity.append(DynatraceConnectivity.ExpiredToken)
+                elif status == 403:
+                    self_monitoring.dynatrace_connectivity.append(DynatraceConnectivity.WrongToken)
+                elif status == 404 or status == 405:
+                    self_monitoring.dynatrace_connectivity.append(DynatraceConnectivity.WrongURL)
+                elif status == 413 or status == 429:
+                    self_monitoring.dynatrace_connectivity.append(DynatraceConnectivity.TooManyRequests)
+                    raise HTTPError(log_ingest_url, 429, "Dynatrace throttling response")
```

Comment:
Did you test this `elif` branch? My IDE marks this HTTPError call as missing 2 additional constructor arguments, and indeed the constructor accepts 5 arguments. The same applies to line `75`, and it

Reply:
You're right 2 arguments are missing - added them (although I was sure I tested it and seemed to be working;) ).